docs/src/main/paradox/server-class.md

```diff
@@ -0,0 +1,117 @@
+
+### Creating a new user
+
+Let us take a look at the source code for the first endpoint, the `/user` URI with a `POST` directive, used to create a new user:
+
+@@snip [QuickstartServer.scala]($g8src$/scala/com/lightbend/akka/http/sample/QuickstartServer.scala) { #user-post }
+
+The snippet above contains a couple of interesting building blocks:
+
+* `path` : matches against the incoming URI, in this case, we want to get all requests that matches `user`.
+* `post` : matches against the incoming Http directive, in this case, we are matching against `POST`.
+* `entity(as[User])` : automatically converts the incoming payload, in this case, we expect JSON, into an entity. We will look more at this functionality in the @ref:[JSON](json.md) section.
+* `complete` : used to reply back to the request. The `StatusCodes.Created` is translated to Http response code 201. We also send back information to the caller in the form of a string.    
+
+When this `Route` is called, we want to create a new user, and we do so by sending a message to the actor `userRegistryActor`. We will look at the implementation of this actor later.
+
+### Retrieving and removing a user
+
+Next we need to define how to retrieve and remove a user, i.e. for the case when the URI `/user/$ID` is used where `$ID` is the id of the user:
+
+@@snip [QuickstartServer.scala]($g8src$/scala/com/lightbend/akka/http/sample/QuickstartServer.scala) { #user-get-delete }
+
+This Route snippet contains a couple of interesting concepts:
+
+* `path("user" / Segment) { => user` : this bit of code matches against URIs of the exact format `/user/$ID` and the `Segment` is automatically extracted into the `user` variable so that we can get to the value passed in the URI. For example `/user/Bruce` will populate the `user` variable with the value "Bruce."
+* `get` : matches against the incoming Http directive.
+
+Let's break down the "business logic" in the first Route:
+
+@@snip [QuickstartServer.scala]($g8src$/scala/com/lightbend/akka/http/sample/QuickstartServer.scala) { #retrieve-user-info }
+
+The code above uses the so-called [ask](http://doc.akka.io/docs/akka/current/scala/actors.html#send-messages) in Akka. This will send a message asynchronously and return a `Future` representing a _possible_ reply. The code above maps the reply to the type `UserInfo`. When the future completes, it will use the second part of the code to evaluate to either `Success`, with or without a result, or a `Failure`. Regardless of the outcome of the future, we should return something to the requester, and we do so by using the `complete` directive with an appropriate response code and value.
+
+The remaining directives used for this route are:
+
+* `~` : fuses `Route`s together - this will become more apparent when you see the complete `Route` definition here below.
```

Comment:
Let's not call it "fuse". Let's change to `concat`. Here's a more complete description of `concat` but we might not want to show that here:

```
`concat`: "concatenates" two or more route alternatives. Routes are attempted one after another. If a route rejects a request the next route in the chain is attempted. This continues until a route in the chain produces a response. If all route alternatives reject the request, the concatenated route rejects the route as well. In that case, route alternatives on the next higher level are attempted. If the root level route rejects the request as well, then an error response is returned that contains information about why the request was rejected.
```

Btw. I think `concat` is the wrong name in Akka Http. We should rename it and call it `or` or something else that emphasizes that this is about alternatives.